src/components/avatar.rs

```diff
@@ -112,49 +114,68 @@ impl Avatar {
         glib::Object::new()
     }
 
-    fn setup_expressions(&self) {
-        let item_expression = Self::this_expression("item");
+    fn create_signal_groups(&self) {
         let imp = self.imp();
+        let user_signal_group = glib::SignalGroup::new(User::static_type());
```

Comment:
I'd like to have the function body like this: 

```rust
        let imp = self.imp();

        let user_signal_group = glib::SignalGroup::new(User::static_type());
        user_signal_group.connect_notify_local(
            Some("type"),
            clone!(@weak self as obj => move |_, _| {
                obj.update_display_name();
                obj.update_avatar();
            }),
        );
        user_signal_group.connect_notify_local(
            Some("first-name"),
            clone!(@weak self as obj => move |_, _| {
                obj.update_display_name();
            }),
        );
        user_signal_group.connect_notify_local(
            Some("last-name"),
            clone!(@weak self as obj => move |_, _| {
                obj.update_display_name();
            }),
        );
        user_signal_group.connect_notify_local(
            Some("avatar"),
            clone!(@weak self as obj => move|_, _| {
                obj.update_avatar();
            }),
        );
        imp.user_signal_group.set(user_signal_group).unwrap();

        let chat_signal_group = glib::SignalGroup::new(Chat::static_type());
        chat_signal_group.connect_notify_local(
            Some("title"),
            clone!(@weak self as obj => move |_, _| {
                obj.update_display_name()
            }),
        );
        chat_signal_group.connect_notify_local(
            Some("avatar"),
            clone!(@weak self as obj => move|_, _| {
                obj.update_avatar();
            }),
        );

        imp.chat_signal_group.set(chat_signal_group).unwrap();
```

- Added and removed some blank lines for better grouping
- Keep the closures as simple as possible (see later suggestions for `update_avatar` function)
- Used hyphen instead of underscore for poperties (dunno if underscore even works, but we use the hyphen by convention)